src/west/app/project.py

```diff
@@ -186,6 +186,10 @@ def do_add_parser(self, parser_adder):
         parser.add_argument('-m', '--manifest-url',
                             help='''manifest repository URL to clone;
                             cannot be combined with -l''')
+        parser.add_argument('-o', '--clone-opt', action='append', default=[],
+                            help='''additional option to pass to 'git clone'
+                            (e.g. '-o=--depth=1'); may be given more than once;
```

Comment:
Just FYI, I tested this PR and compared:
```
1. west init -o=--depth=1
2. west init -o=--filter=tree:0
```

The latter took 30% longer and 10% more disk but has a complete `git log` and does not break `git describe`. We've been using it in CI for a couple years now and it's a much better choice there.